### PR TITLE
Adding support for wildcard transitions.

### PR DESCRIPTION
**Pull request by joaomdmoura**
Related to #17 

This feature will enable people to add transitions that can be
made from any state, pretty useful for use cases where a 'canceled'
state might exist, you can easily declare it as a valid transition from
any state using a wildcard "*" => "next_state"

### CODE REVIEW

File: test/machinery_test.exs [hunk 1]
```diff
@@ -26,6 +26,17 @@ defmodule MachineryTest do
...
+    created_struct = %TestStruct{state: "created", missing_fields: false}
+    partial_struct = %TestStruct{state: "partial", missing_fields: false}
+    completed_struct = %TestStruct{state: "completed"}
+
+    assert {:ok, %TestStruct{state: "canceled", missing_fields: false}} = Machinery.transition_to(created_struct, TestStateMachine, "canceled")
+    assert {:ok, %TestStruct{state: "canceled", missing_fields: false}} = Machinery.transition_to(partial_struct, TestStateMachine, "canceled")
+    assert {:ok, %TestStruct{state: "canceled"}} = Machinery.transition_to(completed_struct, TestStateMachine, "canceled")
+  end
+
+
```
Comment (joaomdmoura) [inline test/machinery_test.exs]: There should be no more than 1 consecutive blank lines.

**Comment by joaomdmoura** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/joaomdmoura/machinery/pulls/32.

File: test/machinery/transition_test.exs [hunk 2]
```diff
@@ -5,12 +5,22 @@ defmodule MachineryTest.TransitionTest do
...
+  test "declared_transition?/3 for a declared transition that allows transition for any state" do
+    transitions = %{
+      "created" => "completed",
+      "*" =>"canceled"
+    }
+    assert Transition.declared_transition?(transitions, "created", "completed")
+    assert Transition.declared_transition?(transitions, "created", "canceled")
+    assert Transition.declared_transition?(transitions, "completed", "canceled")
   end
 end
```
Comment (bragamat) [inline test/machinery/transition_test.exs:26]: about this last "end", is it closing a "do" statement before line 5 or something ?

Comment (joaomdmoura) [reply test/machinery/transition_test.exs:26]: Yup, closing the module itself , the module `MachineryTest.TransitionTes` :)

**Pull request merged**